coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentControllerBase.java

```diff
@@ -354,7 +355,7 @@ protected ModelAndView navigate(ProposalDevelopmentDocumentForm form, BindingRes
          boolean canEdit = form.isCanEditView();
          if (isNavigateAwayFromAttachment(navigateToPageId, form.getPageId())) {
              prepareLocks(form);
-             return narrativePageSave(form, canEdit);
+             return narrativePageSaveOnNavigate(form, canEdit);
```

Comment:
I guess you meant methods narrativePageSave and narrativePageSaveOnNavigate, they are different in, how they handle changed objects. narrativePageSave reads the document and replaces attachments from form object that is good to handle the scenario in which user saves first and navigates away. But it fails to handle the case of user updates/add an attachment and navigates away without saving. narrativePageSaveOnNavigate method handles newly added or updated attachments and don't disturb the already persisted attachments. I think replacing the already persisted attachments (which include blob data) with form data - which may not have the blob data for all the attachments result in attachments getting wiped out.
I am not sure how the locking sections are handled.